validator.go

```diff
@@ -182,22 +190,47 @@ func jsonFieldName(tag reflect.StructTag) string {
 	return jtag
 }
 
+// postTagValidation wraps a validation function, so after the
+// tag validations have been run, checks if the type implements
+func postTagValidation(f ValidationFunc) ValidationFunc {
+	return func(v reflect.Value, state *ErrorReporter) {
+		f(v, state)
+		fmt.Printf("v -> %#v\n\n", v)
+		/*
+			if v.CanAddr() {
+				v = v.Addr()
+				fmt.Printf("Addr -> %#v\n\n", v)
+			}
+		*/
+		i := v.Interface()
+		fmt.Printf("i -> %#v\n\n", i)
+		if i == nil {
+			return
+		}
+		if vi, ok := i.(ValidationInterface); ok {
+			panic("can get a Validation interface")
+			vi.Validate(state)
+		}
+	}
```

Comment:
Follow this convention you can make it with less code :innocent: :
```suggestion
t := reflect.TypeOf(v.Interface())
if t.Implements(reflect.TypeOf((*Validator)(nil))) {
   v.Interface().(ValidationInterface).Validate(state)
}
```